tools/auth/lark.go

```diff
@@ -0,0 +1,82 @@
+package auth
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/pocketbase/pocketbase/tools/types"
+	"golang.org/x/oauth2"
+)
+
+func init() {
+	Providers[NameLark] = wrapFactory(NewLarkProvider)
+}
+
+var _ Provider = (*Lark)(nil)
+
+// NameLark is the unique name of the Lark provider.
+const NameLark string = "lark"
+
+// Lark allows authentication via Lark OAuth2.
+type Lark struct {
+	BaseProvider
+}
+
+// NewLarkProvider creates new Lark provider instance with some defaults.
+func NewLarkProvider() *Lark {
+	return &Lark{BaseProvider{
+		ctx:         context.Background(),
+		displayName: "Lark",
+		pkce:        true,
+		scopes:      []string{"offline_access", "contact:user.employee_id:readonly", "contact:user.email:readonly"},
+		// Lark has two domains with the same API: feishu.cn and larksuite.com.
+		// The former is used in China and the latter is used in the other regions.
+		// We choose feishu.cn as a default, matching the behavier of Lark's official SDK.
+		// Endpoint URLs can be overridden from the frontend if needed.
+		// SDK Reference: https://github.com/larksuite/oapi-sdk-go
+		authURL:     "https://accounts.feishu.cn/open-apis/authen/v1/authorize",
+		tokenURL:    "https://open.feishu.cn/open-apis/authen/v2/oauth/token",
+		userInfoURL: "https://open.feishu.cn/open-apis/authen/v1/user_info",
+	}}
+}
+
+// FetchAuthUser returns an AuthUser instance based the Lark's user api.
+//
+// API reference: https://open.feishu.cn/document/authentication-management/access-token/get-user-access-token
+func (p *Lark) FetchAuthUser(token *oauth2.Token) (*AuthUser, error) {
+	data, err := p.FetchRawUserInfo(token)
+	if err != nil {
+		return nil, err
+	}
+
+	rawUser := map[string]any{}
+	if err := json.Unmarshal(data, &rawUser); err != nil {
+		return nil, err
+	}
+
+	extracted := struct {
+		Data struct {
+			Id        string `json:"user_id"`
+			Name      string `json:"name"`
+			Email     string `json:"email"`
+			AvatarURL string `json:"avatar_url"`
+		} `json:"data"`
+	}{}
+	if err := json.Unmarshal(data, &extracted); err != nil {
+		return nil, err
+	}
+
+	user := &AuthUser{
+		Id:           extracted.Data.Id,
+		Name:         extracted.Data.Name,
+		Email:        extracted.Data.Email,
```

Comment:
There is a warning at the beginning of https://open.feishu.cn/document/server-docs/authentication-management/login-state-management/get:

> Phone numbers and email addresses are contact details imported by the organization administrator and have not been verified by the users in real time. **Developers are advised not to use them directly as login credentials for business systems**. If used, make sure to verify them independently.

PocketBase requires the email address loaded in `AuthUser` to be always verified and since they are not verified by Lark we can't trust it and we should skip the `Email` field assignment.